src/hotspot/share/opto/memnode.cpp

```diff
@@ -4888,10 +4888,8 @@ bool MergeMemStream::match_memory(Node* mem, const MergeMemNode* mm, int idx) {
   if (mem == n)  return true;  // might be empty_memory()
   n = (idx == Compile::AliasIdxBot)? mm->base_memory(): mm->memory_at(idx);
   if (mem == n)  return true;
-  while (n->is_Phi() && (n = n->as_Phi()->is_copy()) != NULL) {
-    if (mem == n)  return true;
-    if (n == NULL)  break;
-  }
+  if (n->is_Phi())
+    assert(!n->as_Phi()->is_copy(), "n cannot be a copy");
```

Comment:
The if condition should be added to the assert:
`assert(!n->is_Phi() || !n->as_Phi()->is_copy() ...`